### PR TITLE
Arangopipe Updates

### PR DESCRIPTION
**Pull request by tjoubert**
- Deprecate Foxx & UI
- Remove Reliance On Managed Service Setup

### CODE REVIEW

File: arangopipe/arangopipe/arangopipe_storage/arangopipe_admin_api.py [hunk 1]
```diff
@@ -388,15 +392,15 @@ def create_enterprise_ml_graph(self, db_replication_factor):
         for edge, fromv, tov in zip(edge_names, from_list, to_list):
             if not self.emlg.has_edge_definition(edge):
```
Comment (geenen124) [inline arangopipe/arangopipe/arangopipe_storage/arangopipe_admin_api.py:393]: we could possibly check if the collection exists independent of if the graph has the edge definition? For example something along the lines of:
```
        if not db.has_collection(edge):
            db.create_collection(
                edge, edge=True, replication_factor=db_replication_factor
            )
        if not emlg.has_edge_definition(edge):
            emlg.create_edge_definition(
                edge_collection=edge,
                from_vertex_collections=[fromv],
                to_vertex_collections=[tov],
            )
```

What do you think?

File: arangopipe/arangopipe/arangopipe_storage/arangopipe_admin_api.py [hunk 2]
```diff
@@ -39,130 +40,144 @@
...
-                self.config.dump_data()
-            logger.error("Obtained connection with retry!")
+        # if self.mscp.DB_ROOT_USER in self.cfg["arangodb"]:
+        #     logger.info("A root user was specified, persisting...")
+
+        # if self.mscp.DB_ROOT_USER_PASSWORD in self.cfg["arangodb"]:
+        #     logger.info("A root user password was specified, persisting...")
+
+        #Provision the graph
+        self.create_enterprise_ml_graph(db_replication_factor)
```
Comment (geenen124) [inline arangopipe/arangopipe/arangopipe_storage/arangopipe_admin_api.py:143]: Would it make sense to add an optional `create_graph` boolean argument to the `__init__` function to make this optional? e.g. `create_graph: bool = True` and then make this provisioning conditional?
```
if create_graph:
    self.create_enterprise_ml_graph(db_replication_factor)
```
(just thinking out loud here)

File: arangopipe/arangopipe/arangopipe_storage/arangopipe_api.py [hunk 3]
```diff
@@ -243,43 +242,41 @@ def lookup_modelperf(self, tag_value):
...
+        #                  str(db_serv_port))
+        # client = ArangoClient(hosts=host_conn_str)
+
+        # self.db = client.db(name=db_name,
+        #                     username=db_user_name,
+        #                     password=db_passwd,
+        #                     verify=True)
 
         if self.db is None:
             return
```
Comment (geenen124) [inline arangopipe/arangopipe/arangopipe_storage/arangopipe_api.py:271]: raise an exception (e.g. a ValueError) here now that we expect this as an input argument?

File: arangopipe/arangopipe/arangopipe_storage/arangopipe_api.py [hunk 4]
```diff
@@ -46,10 +47,10 @@ class ArangoPipe:
     (4) Register you model with ArangoPipe
     """
 
-    def __init__(self, config: ArangoPipeConfig):
+    def __init__(self, db: StandardDatabase, config: ArangoPipeConfig):
         self.cfg = config.get_cfg()
```
Comment (geenen124) [inline arangopipe/arangopipe/arangopipe_storage/arangopipe_api.py:51]: As far as I can tell, with the database now being passed to the `__init__` method here, the only remaining part of the configuration that is still being used - is the enterprise graph name on line 272. We could decide to either:
- Accept the graph name as a string with a default constant value? 
- Or still accept the ArangoPipeConfig object but only store the graph name?
What do you think?

File: arangopipe/arangopipe/arangopipe_storage/arangopipe_api.py [hunk 5]
```diff
@@ -46,10 +47,10 @@ class ArangoPipe:
...
         self.emlg = None
-        self.db = None
+        self.db = db
         self.mscp = ManagedServiceConnParam()
```
Comment (geenen124) [inline arangopipe/arangopipe/arangopipe_storage/arangopipe_api.py:54]: Is this still being used anywhere in this class?

**Review by geenen124 — COMMENTED** [review]
Great work so far! Just added a few comments & questions

File: arangopipe/arangopipe/arangopipe_storage/arangopipe_admin_api.py (hunk 1 above)
Comment (tjoubert) [reply arangopipe/arangopipe/arangopipe_storage/arangopipe_admin_api.py:393]: Very good point

File: arangopipe/arangopipe/arangopipe_storage/arangopipe_admin_api.py (hunk 2 above)
Comment (tjoubert) [reply arangopipe/arangopipe/arangopipe_storage/arangopipe_admin_api.py:143]: Very good point. I've added this.

File: arangopipe/arangopipe/arangopipe_storage/arangopipe_api.py (hunk 3 above)
Comment (tjoubert) [reply arangopipe/arangopipe/arangopipe_storage/arangopipe_api.py:271]: Good point. I've also added this check to the constructor.

File: arangopipe/arangopipe/arangopipe_storage/arangopipe_api.py (hunk 5 above)
Comment (tjoubert) [reply arangopipe/arangopipe/arangopipe_storage/arangopipe_api.py:54]: Yes, methods like `init_graph()` and `lookup_modelperf()` use it.

File: arangopipe/arangopipe/arangopipe_storage/arangopipe_api.py (hunk 4 above)
Comment (tjoubert) [reply arangopipe/arangopipe/arangopipe_storage/arangopipe_api.py:51]: @geenen124, is there anything else that we may put through config in the future? If yes, leaving the `ArangoPipeConfig` with graph name only for now is the best option in my view. Let's discuss on the sync. :-)

File: arangopipe/arangopipe/arangopipe_storage/arangopipe_admin_api.py [hunk 6]
```diff
@@ -40,287 +29,73 @@
...
-            except KeyError:
-                repeated_connection = False
+        self.graph_name = graph_name
+        self.db_replication_factor = replication_factor
+        self.create_graph = create_graph
+        self.reuse_connection = reuse_connection
 
-        return repeated_connection
+        # Provision the graph
+        if self.create_graph:
```
Comment (geenen124) [inline arangopipe/arangopipe/arangopipe_storage/arangopipe_admin_api.py:52]: @tjoubert @cw00dw0rd didn't we want to move this to the `ArangoPipe` class in `arangopipe_api.py`?

**Review by geenen124 — COMMENTED** [review]
Just a few small comments - mainly about which responsibilities we wanted to assign to the `ArangoPipe` class

File: arangopipe/arangopipe/arangopipe_storage/arangopipe_admin_api.py (hunk 6 above)
Comment (tjoubert) [reply arangopipe/arangopipe/arangopipe_storage/arangopipe_admin_api.py:52]: Done. Do we also remove all graph related methods like `remove_edge_definition_from_arangopipe()`, `add_edges_to_arangopipe`, etc...?

**Comment by joerg84** [comment]
Can this be merged @tjoubert @geenen124

**Review by cw00dw0rd — APPROVED** [review]
Basic functionality is still confirmed to be working. However, restructuring, proper testing, and code cleanup are still needed, but IMO can be another PR.

**Comment by cw00dw0rd** [comment]
I am unable to reproduce this error on my local machine, so I am taking shots in the dark, even after ensuring it's using the same version of mypy.

It seems to have some issue with the cursor type, and I have confirmed that the cursor we are attempting to use does have `__iter__` and tried narrowing down where the conflict would be coming from. I suspect this is more of greater codebase/mypy issue, as if I forcibly hardcoded a solution,
https://github.com/arangoml/arangopipe/blob/MLP-87-deprecate-foxx-and-frontend/arangopipe/arangopipe/arangopipe_storage/arangopipe_api.py#L127

 mypy still fails but also doesn't have any errors. 
https://github.com/arangoml/arangopipe/actions/runs/3953235970/jobs/6769280038#step:8:9

I can continue to investigate but considering that we intend to make more substantial changes in the short term I would rather open an issue for this and re-evaluate the code from a higher level.

**Comment by cw00dw0rd** [comment]
Note: If we proceed, I will need to revert some of the hardcoding I did first.

**Comment by geenen124** [comment]
@cw00dw0rd agreed - this looks like a deeper issue - better suited to a separate issue/PR considering the large size+scope of this one already. Let's merge once the hardcoding is reverted 👍

**Pull request merged**